### PR TITLE
Proofs per status endpoint

### PR DESCRIPTION
**Pull request by pettinarip**
Instead of `/proofs`, now we have
- `/proofs/queued`
- `/proofs/proving`
- `/proofs/proved`

### CODE REVIEW

File: app/api/v0/proofs/proving/route.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import { revalidateTag } from "next/cache"
+import { formatGwei } from "viem"
+import { ZodError } from "zod"
+
+import { withAuth } from "@/lib/auth"
+import { fetchBlockData } from "@/lib/blocks"
+import { provingProofSchema } from "@/lib/zod/schemas/proof"
+
+// TODO: refactor code to use baseProofHandler and abstract out the logic
```
Comment (pettinarip) [inline app/api/v0/proofs/proving/route.ts:9]: will be working on this in another PR. At the moment the 3 handlers are pretty much a copy of each other.

**Review by wackerow — APPROVED** [review]
